include/operations/blas2_trees.h

```diff
@@ -424,5 +424,44 @@ GerCol<Single, Lower, Diag, Upper, lhs_t, rhs_1_t, rhs_2_t> make_ger_col(
       lhs_, scalar_, rhs_1_, rhs_2_, nWG_row_, nWG_col_, local_memory_size_);
 }
 
+/**** GERP N COLS x (N + 1)/2 ROWS FOR PACKED MATRIX ****/
+/* This is a specialization of the GER class for the packed
+ * symmetric matrices (P stands for Packed in the name). For more details
+ * on matrix layouts, refer to the explanation here:
+ * https://spec.oneapi.io/versions/1.1-rev-1/elements/oneMKL/source/domains/matrix-storage.html#matrix-storage
+ */
+template <bool Single, bool isUpper, typename lhs_t,
+          typename rhs_1_t, typename rhs_2_t>
+struct Gerp {
+  using value_t = typename rhs_1_t::value_t;
+  using index_t = typename rhs_1_t::index_t;
+
+  lhs_t lhs_;
+  rhs_1_t rhs_1_;
+  rhs_2_t rhs_2_;
+  value_t alpha_;
+  index_t N_, incX_1_, incX_2_;
+
+  Gerp(lhs_t &_l, index_t N_, value_t _alpha, rhs_1_t &_r1, index_t _incX_1,
+       rhs_2_t &_r2, index_t _incX_2);
+  index_t get_size() const;
+  template <int N, bool Upper>
```

Comment:
What is the purpose of `N` template parameter?

Reply:
I added the `N` template parameter to this `compute_row_col` struct to be able to partially specialize it in the code. I wasn't able to fully specialize this struct as I was getting compile time errors when trying to do so.

Reply:
I have removed the `compute_row_col` struct and replaced it with a static function of the same name.